llvm/lib/Transforms/Vectorize/VPlanAnalysis.h

```diff
@@ -63,6 +63,9 @@ class VPTypeAnalysis {
 
   /// Return the LLVMContext used by the analysis.
   LLVMContext &getContext() { return Ctx; }
+
+  /// Remove \p V from the cache. You must call this after a value is erased.
```

Comment:
```suggestion
  /// Remove \p V from the cache. You must call this after a VPValue in the cache is erased.
```

(I think just erasing isn't a problem, the problem is when new VPValues/recipes are created and added to the cache, if they get allocated to the same address as a previous cached VPValue. But that's too detailed ;))